### PR TITLE
Replace promu by goreleaser, replace travis by github actions; Add De…

### PR DESCRIPTION
**Pull request by benjaminguttmann-avtq**
…pendabot

Tried to adapt your changes done to cf_exporter to start streamlining those.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: set up go
+        uses: actions/setup-go@v2
```
Comment (psycofdj) [inline .github/workflows/main.yml:23]: use actions/setup-go@v4

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        with:
+          go-version: 1.18
+
+      - name: cache go modules
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
```
Comment (psycofdj) [inline .github/workflows/main.yml:31]: now builtin setup-go if I'm not mistaken

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            echo "::error::modules are not tidy, please run go mod tidy"
+            exit 1
+          fi
+
+      - name: tests
+        run: |
+          go test -v ./...
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
```
Comment (psycofdj) [inline .github/workflows/main.yml:54]: use docker/login-action@v2

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: run goreleaser
+        uses: goreleaser/goreleaser-action@v2
```
Comment (psycofdj) [inline .github/workflows/main.yml:60]: use goreleaser/goreleaser-action@v4

File: .goreleaser.yml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      - -X github.com/prometheus/common/version.Branch={{.Branch}}
+      - -X github.com/prometheus/common/version.BuildUser={{.Env.USER}}
+      - -X github.com/prometheus/common/version.BuildDate={{.Date}}
+      - -X code.cloudfoundry.org/cli/version/binaryVersion={{.Version}}
+    goos:
+      - linux
+      - darwin
+      - windows
+    goarch:
+      - amd64
```
Comment (psycofdj) [inline .goreleaser.yml:24]: What about dropping all but amd64 architecture ? I always felt like it was a waste of resources to build and upload that many tarballs
If anybody is impacted by this decision, he can easily open an issue and add his favorite arch in the build matrix

File: .goreleaser.yml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      - "boshprometheus/cf-exporter:latest"
+      - "boshprometheus/cf-exporter:{{ .Tag }}"
+      - "boshprometheus/cf-exporter:v{{ .Major }}"
+      - "boshprometheus/cf-exporter:v{{ .Major }}.{{ .Minor }}"
+    build_flag_templates:
+      - "--label=org.opencontainers.image.created={{ .Date }}"
+      - "--label=org.opencontainers.image.title={{ .ProjectName }}"
+      - "--label=org.opencontainers.image.revision={{ .FullCommit }}"
+      - "--label=org.opencontainers.image.version={{ .Version }}"
+      - "--label=org.opencontainers.image.source={{ .GitURL }}"
```
Comment (psycofdj) [inline .goreleaser.yml:93]: need to adapt this part, Dokerfile is located in to root-dir and labels must match bosh_exporter repository, not cf_exporter

**Review by psycofdj — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (benjaminguttmann-avtq) [reply .github/workflows/main.yml:18]: can you point me to where I can find which version to use (for this I took what we had for cf_exporter)

File: .goreleaser.yml (hunk 5 above)
Comment (benjaminguttmann-avtq) [reply .goreleaser.yml:24]: It feels to me that it is the default for prometheus exporter so I would stick to it.

File: .github/workflows/main.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (psycofdj) [reply .github/workflows/main.yml:18]: the marketplace is usually a good entry point: https://github.com/marketplace/actions/checkout

File: .goreleaser.yml (hunk 5 above)
Comment (psycofdj) [reply .goreleaser.yml:24]: :ok_hand:

**Review by psycofdj — CHANGES_REQUESTED** [review]
could you also add a dedicated `tests` workflow  as follow ? It will run on each commit and status will appear on pull requests


- file: `.github/workflows/tests.yml`
- content: 
  ```yaml
  name: tests

  on: push

  jobs:
    tests:
      runs-on: ubuntu-latest
      steps:
        - name: checkout
          uses: actions/checkout@v3
          with:
            fetch-depth: 0
  
        - name: set up go
          uses: actions/setup-go@v4
          with:
            go-version: ">=1.20.0"
  
        - name: running unit-tests
          run: |
            go test -v ./...
  ```

File: .github/workflows/main.yml (hunk 2 above)
Comment (psycofdj) [reply .github/workflows/main.yml:31]: I confirm: setup-go@v4 enables caching by default (see [here](https://github.com/marketplace/actions/setup-go-environment))
image

File: .github/workflows/main.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (benjaminguttmann-avtq) [reply .github/workflows/main.yml:18]: thank you :) will adjust

File: VERSION [hunk 8]
```diff
@@ -1 +0,0 @@
-3.4.0
```
[the remark this answers is not in the document]
Comment (benjaminguttmann-avtq) [reply VERSION:1]: ./dist/binary_darwin_amd64_v1/bosh_exporter --version
fbosh_exporter, version 3.4.0-SNAPSHOT-645f55e (branch: maintance-work, revision: 645f55e594268d056bf6585e348ce9250354c0b3)
  build user:       benjaminguttmann
  build date:       2023-04-04T08:34:40Z
  go version:       go1.17.8
  platform:         darwin/amd64

looks good I would say, no?

File: .github/workflows/main.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (benjaminguttmann-avtq) [reply .github/workflows/main.yml]: adjusted

File: VERSION (hunk 8 above)
[the remark this answers is not in the document]
Comment (psycofdj) [reply VERSION:1]: perfect !

**Review by psycofdj — APPROVED** [review]

**Pull request merged**